### ISSUE TITLE
displaycode does not accept non-ascii characters

### ISSUE THREAD

**Issue by DamienCassou** [cdlm/sbabook#5]
This

``` latex
\begin{displaycode}{smalltalk}
'Ελλάδα'
\end{displaycode}
```

renders are

```
''
```

**Comment by cdlm** [comment]
Yes because everyone knows that characters are bytes always and forever -_-
There is an option for UTF8 support that I want to try…

Le jeudi 3 mars 2016, Damien Cassou notifications@github.com a écrit :

> This
> 
> \begin{displaycode}{smalltalk}
> 'Ελλάδα'\end{displaycode}
> 
> renders are
> 
> ''
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/cdlm/sbabook/issues/5.

## 

Damien Pollet
type less, do more [ | ] http://people.untyped.org/damien.pollet

**Comment by cdlm** [comment]
The real issue is that the font I picked for code has limited support outside Latin alphabets. I'll have to switch to another one. I propose Source Code Pro, but other quality options would be Fira Code or Fira Mono, Hack, Ubuntu Mono…

**Comment by DamienCassou** [comment]
You are the only one who can choose :-). I trust you fully on that.

**Comment by cdlm** [comment]
here's [Hack](http://sourcefoundry.org/hack/). The author added Esperanto support on my request so I'm a bit partial :)

**Comment by DamienCassou** [comment]
Damien Pollet notifications@github.com writes:

> here's [Hack](http://sourcefoundry.org/hack/). The author added Esperanto support on my request so I'm a bit partial :)

I can see "designed for the screen". Is that really appropriate for a book?

## 

Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill

**Comment by cdlm** [comment]
Need to print to see how it looks on paper. I've managed to do some tests: Hack, Fira Mono, Ubuntu Mono work well with accents, greek, and even cyrillic. Strangely, Source Code Pro has the glyphs but they don't show in the PDF, I don't know why. Oh well.

**Comment by cdlm** [comment]
Fixed as of bc2422882f3a10587bfc1c52158c4ce1b7848a2f by switching to Fira Mono, which does work better and has the huge advantage of being distributed in TeXlive.
